pyqtgraph/namedColorManager.py

```diff
@@ -0,0 +1,86 @@
+# from .Qt import QtGui, QtCore, QT_LIB, QtVersion
+from .Qt import QtCore, QtGui
+
+import weakref
+
+__all__ = ['NamedColorManager']
+DEBUG = True
+
+DEFAULT_COLORS = {
+    'b': QtGui.QColor(  0,  0,255,255),
+    'g': QtGui.QColor(  0,255,  0,255),
+    'r': QtGui.QColor(255,  0,  0,255),
+    'c': QtGui.QColor(  0,255,255,255),
+    'm': QtGui.QColor(255,  0,255,255),
+    'y': QtGui.QColor(255,255,  0,255),
+    'k': QtGui.QColor(  0,  0,  0,255),
+    'w': QtGui.QColor(255,255,255,255),
+    'd': QtGui.QColor(150,150,150,255),
+    'l': QtGui.QColor(200,200,200,255),
+    's': QtGui.QColor(100,100,150,255),
+    'gr_acc':QtGui.QColor(200,200,100,255), # graphical accent color: pastel yellow 
+    'gr_reg':QtGui.QColor(  0,  0,255, 50)  # graphical region marker: translucent blue
+}
+for key, col in [ # add functional colors
+    ('gr_fg','d'),  # graphical foreground
+    ('gr_bg','k'),  # graphical background
+    ('gr_txt','d'), # graphical text color
+    ('gr_hov','r')  # graphical hover color
+]:
+    DEFAULT_COLORS[key] = DEFAULT_COLORS[col]
+
+for idx, col in enumerate( ( # twelve predefined plot colors
+    'l','y','r','m','b','c','g','d','d','d','d','d'
+) ): 
+    key = 'p{:X}'.format(idx)
+    DEFAULT_COLORS[key] = DEFAULT_COLORS[col]
+    
+# define and instantiate a SignalSource object to pass signals to all pyqtgraph elements
+class NamedColorManager(QtCore.QObject): # this needs to emit QEvents
```

Comment:
QEvents or Signals?

Reply:
QSignal. Oops. Rewrote the comment a little.